File: exercícios/08-24-beecrowd-1261.py
m, n = map(int, input().split())

hay_points = {}

for _ in range(m):
    nome, price = input().split()
    hay_points[nome] = int(price)

for _ in range(n):
    valor = 0

    while True:
        linha = input()

        if linha == ".":
            break

        palavras = linha.split()

        for palavra in palavras:
            if palavra in hay_points:
                valor += hay_points[palavra]

    print(valor)

File: exercícios/08-24-beecrowd-1261.cpp
#include <iostream>
#include <map>
#include <string>
#include <sstream>

using namespace std;

int main() {
    int m, n;
    cin >> m >> n;

    map<string, int> hay_points;

    for (int i = 0; i < m; i++) {
        string nome;
        int price;

        cin >> nome >> price;

        hay_points[nome] = price;
    }

    string linha;

    getline(cin, linha);

    for (int i = 0; i < n; i++) {
        int valor = 0;

        while (getline(cin, linha) && linha != ".") {
            stringstream ss(linha);
            string palavra;

            while (ss >> palavra) {
                if (hay_points.count(palavra)) {
                    valor += hay_points[palavra];
                }
            }
        }

        cout << valor << endl;
    }

    return 0;
}